src/eos-updater-poll-common.c

```diff
@@ -742,14 +751,59 @@ fetch_latest_commit (OstreeRepo *repo,
                                       error))
     return FALSE;
 
-  return fetch_commit_checksum (repo,
-                                cancellable,
-                                remote_name,
-                                ref,
-                                url_override,
-                                out_checksum,
-                                out_extensions,
-                                error);
+  if (!fetch_commit_checksum (repo,
+                              cancellable,
+                              remote_name,
+                              ref,
+                              url_override,
+                              &checksum,
+                              &extensions,
+                              error))
+    return FALSE;
+
+  if (!ostree_repo_load_variant (repo,
+                                 OSTREE_OBJECT_TYPE_COMMIT,
+                                 checksum,
+                                 &commit,
+                                 error))
+    return FALSE;
+
+  /* If this is a redirect commit, follow it and fetch the new ref instead */
+  g_variant_get_child (commit, 0, "@a{sv}", &metadata);
+  rebase = g_variant_lookup_value (metadata, "ostree.endoflife-rebase", G_VARIANT_TYPE_STRING);
+
+  if (rebase)
```

Comment:
Nitpick: The code style in eos-updater is to be explicit about `NULL` comparisons, so this should be `if (rebase != NULL)`.